Mathlib/Tactic/Propose.lean

```diff
@@ -123,14 +123,17 @@ elab_rules : tactic
         throwError "propose could not find any lemmas using the given hypotheses"
       -- TODO we should have `proposals` return a lazy list, to avoid unnecessary computation here.
       for p in proposals.toList.take 10 do
-        addHaveSuggestion tk (← inferType p.2) p.2
+        addHaveSuggestion tk (← inferType p.2) p.2 (← getRef)
```

Comment:
Maybe you could put `let stx ← getRef` right before the `getMainGoal` line to make it easier to reason about whether `getRef` is getting the right thing.

Reply:
I believe `getRef` reads from the read-only state of the monad, so the only thing that can cause issues is if you use a combinator (with the obvious one to watch out for being `withRef`).